cms/models.py

```diff
@@ -169,6 +170,15 @@ class ProgramIndexPage(CourseObjectIndexPage):
 
     def get_child_by_readable_id(self, readable_id):
         """Fetch a child page by the related Program's readable_id value"""
+        try:
+            queryset = ProgramRun.objects.annotate(
+                program_run=Concat("program__readable_id", models.Value("+"), "run_tag")
```

Comment:
You should be using `courses.constants.ENROLLABLE_ITEM_ID_SEPARATOR` here to join the two values. Don't hardcode the `+` here.

Reply:
it was not aware of it, good to know.

Reply:
@ahmed-arbisoft this doesn't work out. It's treating `+` as a field name and neither converting it into a string.

Reply:
only `models.Value('+')` is working here. because whatever we pass from the variables or formating is being treated as field name except `models.Value('+')`.

Reply:
let me try another hack.